src/base/static/components/organisms/main-map.tsx

```diff
@@ -347,27 +390,23 @@ class MainMap extends React.Component<Props, State> {
     }
   };
 
-  updateMapViewport = throttle(
-    500,
-    (viewport: MapViewportDiff, scrollZoomAroundCenter: boolean): void => {
-      this.props.updateMapViewport(viewport, scrollZoomAroundCenter);
-    },
-  );
+  updateMapViewport = throttle(500, this.props.updateMapViewport);
 
   setMapViewport = (viewport: MapViewportDiff): void => {
     const scrollZoomAroundCenter = this.isMapTransitioning
       ? false
       : this.props.mapConfig.scrollZoomAroundCenter;
+    const newMapViewport = mapViewportReducer(this.state.mapViewport, {
+      type: UPDATE_VIEWPORT,
+      payload: {
+        viewport,
+        scrollZoomAroundCenter,
+      },
+    });
     this.setState({
-      mapViewport: mapViewportReducer(this.state.mapViewport, {
-        type: MAPVIEWPORT_UPDATE,
-        payload: {
-          viewport,
-          scrollZoomAroundCenter,
-        },
-      }),
+      mapViewport: newMapViewport,
```

Comment:
I'm just starting to dig into the new viewport changes... I'm curious why we're still storing the viewport in component state here? Shouldn't we be reading it from the Redux state at this point?

Reply:
>So if I'm understanding your comment from the lat/lng fields PR, component state in MainMap is used to drive the actual viewport consumed by react-map-gl, and Redux sort of "shadows" that viewport state in a throttled way.

Yes, this is correct. The MapGL component needs to render at ~20 frames per second, so I figured it was easiest to keep the viewport in component state for that purpose.

>Is the goal to just make viewport state available to other parts of the app when needed while keeping the perf benefits of component state?

Yep. I was chatting with the maintainer of Redux and this is the approach that he recommended.